storage/local/storage.go

```diff
@@ -510,6 +522,11 @@ func (s *memorySeriesStorage) DropMetricsForFingerprints(fps ...clientmodel.Fing
 
 // Append implements Storage.
 func (s *memorySeriesStorage) Append(sample *clientmodel.Sample) {
+	for ln, lv := range sample.Metric {
+		if len(lv) == 0 {
+			delete(sample.Metric, ln)
+		}
+	}
```

Comment:
This behavior must be documented in the interface. Not only is it important semantics, it also modifies the Sample passed in.